src/test/java/com/pretz/dyntreeserver/controller/DynTreeValidationTest.java

```diff
@@ -0,0 +1,271 @@
+
+        assertFalse(violations.isEmpty());
+        assertEquals(1, violations.size());
+        ConstraintViolation violation = violations.iterator().next();
+        assertEquals("must not be blank", violation.getMessage());
+        assertEquals("familyName", violation.getPropertyPath().toString());
+        assertEquals("    ", violation.getInvalidValue());
+    }
+
+    @Test
+    public void shouldFailValidationEmptyFamilyName() {
+        testDynTreeDTO = DynTreeDTO.builder()
+                .familyName("")
+                .familyCount(10)
+                .generationsCount(3)
+                .childrenPerCharacter(2.2)
+                .nameListId(4L)
+                .maxAge(90)
+                .maturityAge(18)
+                .startingYear(Year.of(1950))
+                .mainCharacterName("Andrzej")
+                .build();
+
+        Set<ConstraintViolation<DynTreeDTO>> violations
+                = validator.validate(testDynTreeDTO);
+
+        assertFalse(violations.isEmpty());
+        assertEquals(1, violations.size());
+        ConstraintViolation violation = violations.iterator().next();
+        assertEquals("must not be blank", violation.getMessage());
+        assertEquals("familyName", violation.getPropertyPath().toString());
+        assertEquals("", violation.getInvalidValue());
+    }
+
+    @Test
+    public void shouldFailValidationNullFamilyName() {
+        testDynTreeDTO = DynTreeDTO.builder()
+                .familyName(null)
+                .familyCount(10)
+                .generationsCount(3)
+                .childrenPerCharacter(2.2)
+                .nameListId(4L)
+                .maxAge(90)
+                .maturityAge(18)
+                .startingYear(Year.of(1950))
+                .mainCharacterName("Andrzej")
+                .build();
+
+        Set<ConstraintViolation<DynTreeDTO>> violations
+                = validator.validate(testDynTreeDTO);
+
+        assertFalse(violations.isEmpty());
+        assertEquals(1, violations.size());
+        ConstraintViolation violation = violations.iterator().next();
+        assertEquals("must not be blank", violation.getMessage());
+        assertEquals("familyName", violation.getPropertyPath().toString());
+        assertNull(violation.getInvalidValue());
+    }
+
+    @Test
+    public void shouldFailValidationFamilyCountTooSmall() {
+        testDynTreeDTO = DynTreeDTO.builder()
+                .familyName("Duda")
+                .familyCount(1)
+                .generationsCount(3)
+                .childrenPerCharacter(2.2)
+                .nameListId(4L)
+                .maxAge(90)
+                .maturityAge(18)
+                .startingYear(Year.of(1950))
+                .mainCharacterName("Andrzej")
+                .build();
+
+        Set<ConstraintViolation<DynTreeDTO>> violations
+                = validator.validate(testDynTreeDTO);
+
+        assertFalse(violations.isEmpty());
+        assertEquals(1, violations.size());
```

Comment:
I'd say this assert includes the one above

Reply:
Yes, but there is no downside to check both conditions I believe

Reply:
One major downside - lower readability. One reads stuff and needs to wonder if there is some "hidden" difference between these two or not.

Good code is super explicit and clear code